### DISCUSSION TITLE
node listen to ipv6 only

### DISCUSSION THREAD

**Discussion by Comandante1977** [outline/outline#3670] [Self Hosting]
Hi!

Installed everything and trying to get it work. But after `yarn start` I can see, that node is listen only on IPv6 interface.
![image](https://user-images.githubusercontent.com/45628795/174400756-c0aa0051-b5d9-4a6b-8334-d1da947dbda2.png)
How I can change it to ipv4 interface?

**Comment by almereyda** [comment]
It depends on the configuration of your system. The `sysctl` setting `net.ipv6.bindv6only` controls how listening on `::` is interpreted with regards to dual stack mode of IPv6 and IPv4, for one part. (ref. https://github.com/nodejs/node/issues/9390)

Could you verify that the daemon is not reachable at `127.0.0.1:3000` by issuing

    curl --ipv4 localhost:3000

This is also very much related to #3666, that depends on the process listening on `localhost` through

https://github.com/outline/outline/blob/2a6d6f58042b477b0c175a95b554066fced6a572/server/index.ts#L73-L77

without having that configurable. Thus it appears your system interprets `localhost` as `[::]` and is not configured for dual stack use.

**Reply by Comandante1977** [reply]
It became to work in anyway, but another problems - nodemailer not working, cannot upload files, cannot edit!!!!
Wonderful software with zero documentation

**Reply by almereyda** [reply]
Would you like to open new discussions regarding the other follow up subjects, as you did with Configuring the SMTP MDA for transactional mail exchange in #3676? We could then investigate them in further detail there.

Maybe the documentation about

- configuring the S3 storage for file uploads  
  https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/aws-s3-N4M0T6Ypu7
  - #3307 
  - https://github.com/outline/outline/discussions/3281#discussioncomment-2430516

can give a little detail about what else may cause irritations, and help to investigate that subsequently.

Next, one could test for WebSockets to debug editing. They need to be properly proxied to the application, in order to work. And, do you use collaborative editing, and have it enabled for your instance?

- https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/nginx-6htaRboR57
  - https://github.com/outline/outline/discussions/2222#discussioncomment-864452

Maybe this guide, which exists now, also helps to shed some light on the inner workings of the application?

- https://thomasgriffin.com/how-to-install-the-outline-knowledge-base-wiki-on-ubuntu/
